python/API_GW_CUSTOMDOMAIN_CHECK/API_GW_CUSTOMDOMAIN_CHECK.py

```diff
@@ -0,0 +1,395 @@
+# Copyright 2017-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+
+import json
+import sys
+import datetime
+import boto3
+import botocore
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::ApiGateway::DomainName'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+#############
+# Main Code #
+#############
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    apigw = get_client('apigateway', event)
+    customdomains = []
+    evaluations = []
+    count = 0
+    # make the api call
+    response = apigw.get_domain_names()
+    # check whether there is a custom domain name
+    if not response['items']:
+        restapigws = apigw.get_rest_apis()
+        for count in range(len(restapigws['items'])):
+            print(restapigws['items'][count]['id'] + " API gateway does not have a custom domain name.\n")
+            evaluations.append(build_evaluation(restapigws['items'][count]['id'], 'NON_COMPLIANT', event, annotation='Your API Gateway does not have a custom domain name'))
+    else:
+        customdomains += response['items']
+        for each_customdomain in customdomains:
+            # print custom domain to Cloudwatch logs
+            print(each_customdomain)
+            if each_customdomain['domainName']:
+                evaluations.append(build_evaluation(each_customdomain['regionalDomainName'], 'COMPLIANT', event, annotation='This API Gateway has a custom domain name'))
+    return evaluations
+
+def evaluate_parameters(rule_parameters):
+    """Evaluate the rule parameters dictionary validity. Raise a ValueError for invalid parameters.
+
+    Return:
+    anything suitable for the evaluate_compliance()
+
+    Keyword arguments:
+    rule_parameters -- the Key/Value dictionary of the Config Rules parameters
+    """
+    valid_rule_parameters = rule_parameters
```

Comment:
You might consider to have a parameter to check if the custom domain is correct. Something like a list of regex patterns